domain/src/main/java/edu/nextstep/camp/calculator/Calculator.kt

```diff
@@ -1,4 +1,72 @@
 package edu.nextstep.camp.calculator
 
+import java.util.Stack
+
 class Calculator {
+
+    fun evaluate(expression: String?): Int {
+        val stack = Stack<String>()
+        val expressionWithoutBlank = expression?.split(BLANK) ?: throw IllegalArgumentException()
+
+        expressionWithoutBlank.forEach {
+            if (stack.isEmpty()) {
+                stack.push(it)
+            } else {
+                pushValueWhenStackIsNotEmpty(stack, it)
+            }
+        }
+        return if (stack.isEmpty()) throw IllegalArgumentException() else stack.pop().toInt()
+    }
+
+    private fun pushValueWhenStackIsNotEmpty(stack: Stack<String>, str: String) {
+        if (isMathematicalSymbol(str)) {
+            stack.push(str)
+        } else {
+            val type = stack.pop()
+            val value = stack.pop()
+            val result = calculateValue(type, value.toInt(), str.toInt())
+            stack.push(result.toString())
+        }
+    }
+
+    private fun isMathematicalSymbol(symbol: String): Boolean {
+        when(symbol) {
+            MathematicalSymbol.PLUS.type -> return true
+            MathematicalSymbol.MINUS.type -> return true
+            MathematicalSymbol.MULTIPLE.type -> return true
+            MathematicalSymbol.DIVIDE.type -> return true
+        }
+        return false
+    }
```

Comment:
코틀린에는 좋은 컬렉션 API들이 있습니다. 
여러가지 방법이 있겠지만 이런 표준 API들로 간결한 표현식을 만들어서 사용해보세요. ✌

```suggestion
    private fun isMathematicalSymbol(symbol: String): Boolean {
        return MathematicalSymbol.values().any { it.type == symbol }
    }
```

Reply:
오 이런 간결한 컬렉션 API를 알려주셔서 감사합니다 코드가 확 줄어드니깐 좋습니다! 👍